internal/component/pyroscope/java/loop.go

```diff
@@ -23,7 +25,12 @@ import (
 	gopsutil "github.com/shirou/gopsutil/v3/process"
 )
 
-const spyName = "alloy.java"
+const (
+	spyName       = "alloy.java"
+	processJfrDir = "/tmp"
```

Comment:
It's odd that the process has been using `/tmp` for this. It'd make more sense to use the directory specified by the `--storage.path` [cmd arg](https://grafana.com/docs/alloy/latest/reference/cli/run/), similarly to other processes such as `loki.file` and `prometheus.remote_write`. 

@grafana/grafana-alloy-profiling-maintainers Would you mind if we change `pyroscope.java` to use the storage path instead please? I don't feel comfortable with Alloy deleting anything outside of that directory.

Note that if we do this we'd have to update the [docs](https://github.com/grafana/alloy/blob/main/docs/sources/reference/components/pyroscope/pyroscope.java.md) too. They currently [mention](https://grafana.com/docs/alloy/latest/reference/components/pyroscope/pyroscope.java/#profiling-behavior) that `tmp_dir` is used. The docs also mention this:

> The asprof binary runs with root permissions. If you change the tmp_dir configuration to something other than /tmp, then you must ensure that the directory is only writable by root.

I suppose using the storage path still won't be a problem, since root should also have access to it? IDK why the docs say "**only** writable by root" though.

Reply:
Hi @ptodev, thanks for reviewing

> It's odd that the process has been using /tmp for this. It'd make more sense to use the directory specified by the --storage.path [cmd arg](https://grafana.com/docs/alloy/latest/reference/cli/run/)

Maybe this was to distinguish the storage directory on alloy containers from a directory on the profiled java process's container. `tmp_dir` needs to be a directory on the java container for async-profiler to work

I can change the JFR files to be written to `tmp_dir` as well though, since right now only the async-profiler binaries are written to that path

Reply:
> Hi, thank you for the PR! I'm worried about deleting files in directories which Alloy doesn't "own", so it'd be nice if we use the --storage.path dir instead.

Another option could be changing the file name to something less likely to conflict, like by prepending `alloy-`